autofit/optimize/non_linear/nested_sampling/dynesty.py

```diff
@@ -0,0 +1,527 @@
+
+        Parameters
+        ----------
+        paths : af.Paths
+            A class that manages all paths, e.g. where the phase outputs are stored, the non-linear search chains,
+            backups, etc.
+        sigma : float
+            The error-bound value that linked Gaussian prior withs are computed using. For example, if sigma=3.0,
+            parameters will use Gaussian Priors with widths coresponding to errors estimated at 3 sigma confidence.
+        terminate_at_acceptance_ratio : bool
+            If *True*, the sampler will automatically terminate when the acceptance ratio falls behind an input
+            threshold value (see *NestedSampler* for a full description of this feature).
+        acceptance_ratio_threshold : float
+            The acceptance ratio threshold below which sampling terminates if *terminate_at_acceptance_ratio* is
+            *True* (see *NestedSampler* for a full description of this feature).
+
+        Attributes
+        ----------
+        sigma : float
+            The error-bound value that linked Gaussian prior withs are computed using. For example, if sigma=3.0,
+            parameters will use Gaussian Priors with widths coresponding to errors estimated at 3 sigma confidence.
+        iterations_per_update : int
+            The number of iterations performed between every Dynesty back-up (via dumping the Dynesty instance as a
+            pickle).
+        """
+
+        super().__init__(paths=paths, sigma=sigma)
+
+        self.iterations_per_update = self.config("iterations_per_update", int)
+        self.bound = self.config("bound", str)
+        self.sample = self.config("sample", str)
+        self.bootstrap = self.config("bootstrap", int)
+        self.enlarge = self.config("enlarge", float)
+
+        self.update_interval = self.config("update_interval", float)
+
+        if self.update_interval < 0.0:
+            self.update_interval = None
+
+        if self.enlarge < 0.0:
+            if self.bootstrap == 0.0:
+                self.enlarge = 1.0
+            else:
+                self.enlarge = 1.25
+
+        self.vol_dec = self.config("vol_dec", float)
+        self.vol_check = self.config("vol_check", float)
+        self.walks = self.config("walks", int)
+        self.facc = self.config("facc", float)
+        self.slices = self.config("slices", int)
+        self.fmove = self.config("fmove", float)
+        self.max_move = self.config("max_move", int)
+
+        logger.debug("Creating DynestyStatic NLO")
+
+    def copy_with_name_extension(self, extension, remove_phase_tag=False):
+        """Copy this instance of the dynesty non-linear search with all associated attributes.
+
+        This is used to set up the non-linear search on phase extensions."""
+        copy = super().copy_with_name_extension(
+            extension=extension, remove_phase_tag=remove_phase_tag
+        )
+        copy.iterations_per_update = self.iterations_per_update
```

Comment:
We definitely can do better than this. Feels very clunky manually passing all the attributes from one to the other

Reply:
Agree - worth sorting next week as we gear up for more samplers.